src/interpreter/compiler.ts

```diff
@@ -419,10 +432,16 @@ const compilers: CompilerMapping = {
 
     jumpInstr.instrAddress = instructions.length;
 
-    const assignInstr = constructAssignInstr(
-      getSymbolTableEntry(getNameFromDeclaratorPattern(node.id), symbolTable),
-      1
+    const symbolTableEntry = getSymbolTableEntry(
+      getNameFromDeclaratorPattern(node.id),
+      symbolTable
     );
+    if (isBuiltinFunctionSymbolTableEntry(symbolTableEntry)) {
```

Comment:
If we're checking whether the name belongs to a user declared function, would it be more comprehensive to check the symbol table entry type is `Function`?

Reply:
Oh yea, that's a good idea.